src/RA_Defs.cpp

```diff
@@ -1,68 +1,87 @@
 #include "RA_Defs.h"
 
-#include <stdio.h>
-#include <Windows.h>
-#include <locale>
-#include <codecvt>
+#include <iomanip>
+
 
 GetParseErrorFunc GetJSONParseErrorStr = GetParseError_En;
+namespace ra {
 
-static_assert(sizeof(BYTE*) == sizeof(char*), "dangerous cast ahead");
-char* DataStreamAsString(DataStream& stream)
-{
-	return reinterpret_cast<char*>(stream.data());
-}
 
-std::string Narrow(const wchar_t* wstr)
-{
-	static std::wstring_convert< std::codecvt_utf8_utf16< wchar_t >, wchar_t > converter;
-	return converter.to_bytes(wstr);
+
+std::string DataStreamAsString(const DataStream& stream) {
+
+    auto str{ convert_string<std::string>(stream) };
+
+    // pesky null character
+    if (!str.empty()) {
+        str.pop_back();
+    }
+
+    return str; // ok it's definitly showing how it should be
 }
 
-std::string Narrow(const std::wstring& wstr)
-{
-	static std::wstring_convert< std::codecvt_utf8_utf16< wchar_t >, wchar_t > converter;
-	return converter.to_bytes(wstr);
+
+std::string Narrow(const std::wstring& wstr) {
+    return convert_string<std::string>(wstr);
```

Comment:
This doesn't work. If passed L"TEST", it returns "84698384". The "oss << i" treats each wchar_t as an integer.

Reply:
OK, thanks for letting me know. It seemed to work for directories and web response parsing. I will look again.